.github/workflows/CI.yml

```diff
@@ -0,0 +1,26 @@
+name: CI
+
+on: 
+  pull_request:
+    types:
+      - opened
+      - edited
+      - synchronize
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  Run-Tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+        with:
+          enable-cache: true
+
```

Comment:
I believe that you will have to add a step to run `uv sync --frozen`, although it is possible that that is done during the previous step.